terminal/bundles/org.eclipse.terminal.view.ui/src/org/eclipse/terminal/view/ui/launcher/ILaunchDelegateManager.java

```diff
@@ -22,9 +25,9 @@ public interface ILaunchDelegateManager {
 	 * @param unique If <code>true</code>, the method returns new instances for each
 	 *               contributed terminal launcher delegate.
 	 *
-	 * @return The list of contributed terminal launcher delegates, or an empty array.
+	 * @return The list of contributed terminal launcher delegates, or an empty list.
 	 */
-	ILauncherDelegate[] getLauncherDelegates(boolean unique);
+	List<ILauncherDelegate> getLauncherDelegates(boolean unique);
```

Comment:
I would rather use

```suggestion
	Stream<ILauncherDelegate> getLauncherDelegates(boolean unique);
```

instead of List what would require to document otherwise if the returned list is a copy and can be modified by the caller.

Reply:
> I was scared by sibling method, that is asked for size of list and for element at specific index.

I think this is more because of the previous array nature of return type than really required.

If we look for example at `LocalLauncherHandler` there is actually a filtering happen and then the first match is chosen, something that can easily be expressed with `Stream.filter(...).findFirst()`

Then in `PropertyTester` it is checked for empty what can become `Stream.findFirst().isPresent()`

`LaunchTerminalSettingsDialog` also performs a filtering and then a mapping ...

`LaunchTerminalCommandHandler`  is the only place where we actually only want to know if we can omit the dialog, here one can easily collect a List (or even array as before).

I therefore think `Stream<..>` instead of list should be used here.